lib/features/settings/chat_language/service/chat_language_service.dart

```diff
@@ -0,0 +1,19 @@
+import 'dart:convert';
+
+import 'package:cloud_functions/cloud_functions.dart';
+import 'package:public_chat/features/settings/chat_language/data/chat_language.dart';
+
+class ChatLanguageService {
+  static final FirebaseFunctions _functions = FirebaseFunctions.instance;
+
+  static Future<List<ChatLanguage>> getSupportedLanguages() async {
+    try {
+      final result =
+          await _functions.httpsCallable('getSupportedLanguages').call();
+      final List<dynamic> json = jsonDecode(jsonEncode(result.data));
```

Comment:
why do u need to encode a json, then decode it at the same time?

Reply:
I already use cloud function before. Sometime I has issue relate json output type like this [issue](https://github.com/firebase/flutterfire/issues/4059) . So I need decode and encode same time to correct json type. It's just a habit when working with cloud functions to make sure there are no bugs.